dataland-frontend/src/components/forms/parts/elements/basic/UploadDocumentsForm.vue

```diff
@@ -102,32 +103,43 @@ export default defineComponent({
      * @param event full event object containing the files
      * @param event.files files
      */
-    async handleFilesSelected(event: FileUploadSelectEvent): Promise<void> {
+    handleFilesSelected(event: FileUploadSelectEvent) {
       const selectedFilesByUser = event.files as File[];
       if (isThereActuallyANewFileSelected(selectedFilesByUser, this.documentsToUpload)) {
-        const lastSelectedFile = selectedFilesByUser[selectedFilesByUser.length - 1];
-        const documentToUpload = { file: lastSelectedFile } as DocumentToUpload;
-        documentToUpload.reference = await calculateSha256HashFromFile(documentToUpload.file);
-        documentToUpload.fileNameWithoutSuffix = removeFileTypeExtension(documentToUpload.file.name);
-        this.documentsToUpload.push(documentToUpload);
-        this.emitDocumentsChangedEvent();
+        const documentsToUpload = Promise.all(
+          selectedFilesByUser.map(async (file) => {
+            return {
+              file: file,
+              reference: await calculateSha256HashFromFile(file),
+              fileNameWithoutSuffix: removeFileTypeExtension(file.name),
+            };
+          }),
+        ) as Promise<DocumentToUpload[]>;
+
+        void documentsToUpload.then((documentsToUpload) => {
+          this.documentsToUpload = documentsToUpload;
+          this.emitReportsUpdatedEvent();
+        });
       }
     },
     /**
      * Emits event that selected documents changed
      */
-    emitDocumentsChangedEvent() {
-      this.$emit("documentsChanged", this.documentsToUpload);
+    emitReportsUpdatedEvent() {
+      this.$emit("reportsUpdated", this.documentsToUpload);
     },
 
     /**
-     * Remove document from files uploaded
-     * @param indexOfFileToRemove index number of the file to remove
+     * Remove documents from files uploaded
+     * @param indexesOfFilesToRemove index list of numbers of the files to remove
      */
-    removeDocumentFromDocumentsToUpload(indexOfFileToRemove: number) {
-      ((this.$refs.fileUpload as FileUpload).remove as (index: number) => void)(indexOfFileToRemove);
-      this.documentsToUpload.splice(indexOfFileToRemove, 1);
-      this.emitDocumentsChangedEvent();
+    removeDocumentsFromDocumentsToUpload(indexesOfFilesToRemove: number[]) {
+      const sortedAndUniqueIndexes = [...new Set(indexesOfFilesToRemove.sort((a, b) => b - a))];
```

Comment:
Why not just `indexesOfFilesToRemove.reverse()` ?

Reply:
When a user adds multiple files, we don't know which ones are the duplicates until everything is parsed. Theoretically, they might be since we're looping through them, but this both sorts and reverses in one go... just in case.

Reply:
Ah okay.  So `duplicatesWithIndex` is probably already sorted ascending (because looping through a list normally starts with the frist element and continues until the last).  Normally you could just reverse it and it would be fine.
But to be 100% sure, you additionally sort it.

Correct?

That would be fine for me.

Reply:
I committed it with .reverse().
Lets see if something breaks.

Reply:
I will revert it as this was a try to fix sonar code smells

Reply:
Hrm... I'll look into it.

Reply:
Fixed.